staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -628,3 +611,27 @@ func (s *SecureServingInfo) HostPort() (string, int, error) {
 	}
 	return host, port, nil
 }
+
+// AuthorizeClientBearerToken wraps the authenticator and authorizer in loopback authentication logic
+// if the loopback client config is specified AND it has a bearer token.
+func AuthorizeClientBearerToken(loopback *restclient.Config, authn *AuthenticationInfo, authz *AuthorizationInfo) {
+	if loopback == nil || authn == nil || authz == nil {
+		return
+	}
+	if authn.Authenticator != nil && authz.Authorizer != nil && loopback != nil && len(loopback.BearerToken) > 0 {
```

Comment:
then this if is extraneous

Reply:
inverse of your variant above. merge both clauses.